tsfile/src/main/java/org/apache/iotdb/tsfile/read/query/timegenerator/node/AndNode.java

```diff
@@ -45,42 +50,78 @@ public boolean hasNext() throws IOException {
     if (hasCachedValue) {
       return true;
     }
-    if (leftChild.hasNext() && rightChild.hasNext()) {
-      long leftValue = leftChild.next();
-      long rightValue = rightChild.next();
-      while (true) {
-        if (leftValue == rightValue) {
-          this.hasCachedValue = true;
-          this.cachedValue = leftValue;
-          return true;
-        } else if (leftValue > rightValue) {
-          if (rightChild.hasNext()) {
-            rightValue = rightChild.next();
-          } else {
-            return false;
-          }
-        } else { // leftValue < rightValue
-          if (leftChild.hasNext()) {
-            leftValue = leftChild.next();
-          } else {
-            return false;
-          }
+    cachedValue = new TimeSeries(1000);
+    //fill data
+    fillLeftData();
+    fillRightData();
+    /*
+     *  [1,2,3,4,5]   <-   that was stopBatchTime mean
+     *  [1,2,3,4,5,6]
+     */
+    long stopBatchTime = getStopBatchTime();
+
+    while (leftPageData.hasMoreData() && rightPageData.hasMoreData()) {
+      long leftValue = leftPageData.currentTime();
+      long rightValue = rightPageData.currentTime();
+      if (leftValue == rightValue) {
+        this.hasCachedValue = true;
+        this.cachedValue.add(leftValue);
+        leftPageData.next();
+        rightPageData.next();
+      } else if (leftValue > rightValue) {
+        rightPageData.next();
+      } else { // leftValue < rightValue
+        leftPageData.next();
+      }
+
+      if (leftValue > stopBatchTime && rightValue > stopBatchTime) {
```

Comment:
if stop time is 4, the time 5 seems could be cached at this round

Reply:
yes , it's bug ,i will fix